### PR TITLE
feat: add tissue_type and new validation rules for tissue_ontology_term_id and cell_type_ontology_term_id

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
#514 
#517 

Changes: 
- added categorical field 'tissue_type' as required curator-annotated field, enforcing value as one of either 'organoid', 'cell culture', or 'tissue'
- enforced new cell_type_ontology_term_id rules (i.e. MUST be CL term but must NOT be one of the listed forbidden terms)
- enforced new tissue_type_ontology_term_id rules
               - do NOT accept suffixes in term_ids and do not append them to the resulting label
               - enforce cell_type_ontology_term_id rules on this term_id IF tissue_type is 'cell culture', 
               - enforce term_id is child term of [UBERON:0001062](https://www.ebi.ac.uk/ols4/ontologies/uberon/classes?obo_id=UBERON%3A0001062) if tissue_type is 'organoid' or 'tissue'
- simplified parts of code that enforced term_id suffix logic (in write-labels and validate modules), as they no longer apply to any terms and we have no plans to support them again
- removed validation of schema definition in 'validate' command (this is already handled by unit tests, we shouldn't expose code validation logic into a user / deployment facing function)
- fixed tests enforcing schema definition (were not being run previously due to a bad if check) and updated them to fit current schema definition structure
- added tests + update h5ad fixtures to contain tissue_type column w/ appropriate values

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 1]
```diff
@@ -800,6 +761,13 @@ def _validate_seurat_convertibility(self):
                     )
 
                 self.is_seurat_convertible = False
+        if self.adata.raw and self.adata.raw.X.shape[1] != self.adata.raw.var.shape[0]:
```
Comment (nayib-jose-gloria) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:764]: this is in main already, not sure why its showing up as a diff...

File: cellxgene_schema_cli/tests/test_schema_compliance.py [hunk 2]
```diff
@@ -281,16 +281,31 @@ def test_assay_ontology_term_id(self):
 
     def test_cell_type_ontology_term_id(self):
         """
-        cell_type_ontology_term_id categorical with str categories. This MUST be a CL term.
+        cell_type_ontology_term_id categorical with str categories. This MUST be a CL term, and must NOT match forbidden
```
Comment (Bento007) [inline cellxgene_schema_cli/tests/test_schema_compliance.py:284]: Do we have a locally glossary somewhere for these acronyms(CL)?

File: cellxgene_schema_cli/tests/test_schema_compliance.py [hunk 3]
```diff
@@ -281,16 +281,31 @@ def test_assay_ontology_term_id(self):
...
         # Not a valid term
-        self.validator.adata.obs.loc[self.validator.adata.obs.index[0], "cell_type_ontology_term_id"] = "EFO:0000001"
-        self.validator.validate_adata()
-        self.assertEqual(
-            self.validator.errors,
-            ["ERROR: 'EFO:0000001' in 'cell_type_ontology_term_id' is not a valid " "ontology term id of 'CL'."],
-        )
+        with self.subTest(forbidden_term="EFO:0000001"):
+            self.validator.adata.obs.loc[
+                self.validator.adata.obs.index[0], "cell_type_ontology_term_id"
```
Comment (Bento007) [inline cellxgene_schema_cli/tests/test_schema_compliance.py:291]: What is the benefit of using the numpy method of slicing `foo[ x, y]`, as opposed to pythons way using `foo[x][y]`?

**Review by Bento007 — APPROVED** [review]

File: cellxgene_schema_cli/tests/test_schema_compliance.py (hunk 2 above)
Comment (nayib-jose-gloria) [reply cellxgene_schema_cli/tests/test_schema_compliance.py:284]: I don't think its an acronym, its a prefix used for cell ontology terms. There's a glossary in the user-facing [schema](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md#required-ontologies) documentation

File: cellxgene_schema_cli/tests/test_schema_compliance.py (hunk 3 above)
Comment (nayib-jose-gloria) [reply cellxgene_schema_cli/tests/test_schema_compliance.py:291]: this is the syntax for the [loc](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html) accessor function for pandas dataframes

**Pull request merged**